anymail/backends/mailjet.py

```diff
@@ -0,0 +1,124 @@
+from ..exceptions import AnymailRequestsAPIError
+from ..message import AnymailRecipientStatus, ANYMAIL_STATUSES
+from ..utils import get_anymail_setting
+
+from .base_requests import AnymailRequestsBackend, RequestsPayload
+
+
+class EmailBackend(AnymailRequestsBackend):
+    """
+    Mailjet API Email Backend
+    """
+
+    esp_name = "Mailjet"
+
+    def __init__(self, **kwargs):
+        """Init options from Django settings"""
+        esp_name = self.esp_name
+        self.api_key = get_anymail_setting('api_key', esp_name=esp_name, kwargs=kwargs, allow_bare=True)
+        api_url = get_anymail_setting('api_url', esp_name=esp_name, kwargs=kwargs,
+                                      default="https://api.mailjet.com/v3")
+        if not api_url.endswith("/"):
+            api_url += "/"
+        super(EmailBackend, self).__init__(api_url, **kwargs)
+
+    def build_message_payload(self, message, defaults):
+        return MailjetPayload(message, defaults, self)
+
+    def parse_recipient_status(self, response, payload, message):
+        parsed_response = self.deserialize_json_response(response, payload, message)
+        recipient_status = {}
+        try:
+            for key in parsed_response:
+                status = key.lower()
+                if status not in ANYMAIL_STATUSES:
+                    status = 'unknown'
+
+                for item in parsed_response[key]:
+                    message_id = item.get('MessageID_id', None)
+                    email = item['Email']
+                    recipient_status[email] = AnymailRecipientStatus(message_id=message_id, status=status)
+        except (KeyError, TypeError):
+            raise AnymailRequestsAPIError("Invalid Mailjet API response format",
+                                          email_message=message, payload=payload, response=response,
+                                          backend=self)
+        return recipient_status
+
+
+class MailjetPayload(RequestsPayload):
+
+    def __init__(self, *args, **kwargs):
+        self.esp_extra = {}  # late-bound in serialize_data
+        self.headers = {
+            'Content-Type': 'application/json',
+        }
+        super(MailjetPayload, self).__init__(*args, **kwargs)
+
+    def get_api_endpoint(self):
+        return "messages/send"
+
+    def serialize_data(self):
+        return self.serialize_json(self.data)
+
+    #
+    # Payload construction
+    #
+
+    def init_payload(self):
+        self.data = {
+        }
+
+    def set_from_email(self, email):
+        self.data["FromEmail"] = email.email
+        if email.name:
+            self.data["FromName"] = email.name
+
+    def add_recipient(self, recipient_type, email):
+        assert recipient_type in ["to", "cc", "bcc"]
+        if email.name:
+            address = "{} <{}>".format(email.name, email.email)
```

Comment:
This *might* have character-escaping issues (if, say, name contains angle brackets or commas).

If the goal is a standard RFC-formatted name/email pair, you can just use `email.address` for both cases. It returns a properly formatted, quoted, and escaped recipient -- e.g., `"Edmunds, Mike" <medmunds@example.com>`. (Haven't checked Mailjet's docs to see what they expect.)